File: rag-backend/src/models/enums.py
from enum import Enum

class InformationTier(Enum):
    TITLE_ONLY = "title_only"
    TITLE_DESC = "title_desc"
    # use metadata as text for embeddings
    MaT_TITLE_ONLY = "mat_title_only"
    MaT_TITLE_DESC = "mat_title_desc"

class RagPipeline(Enum):
    HYBRID = "hybrid"
    GRAPH = "graph"
    FUSION = "fusion"

class QuestionVariant(Enum):
    NORMAL = "normal"
    ABSTRACT = "abstract"
    DETAILED = "detailed"

File: rag-web-ui/src/types/enums.ts

export enum QuestionVariant {
    NORMAL = "normal",
    ABSTRACT = "abstract",
    DETAILED = "detailed"
}

export enum InformationTier {
    TITLE_ONLY = "title_only",
    TITLE_DESC = "title_desc",
    MaT_TITLE_ONLY = "mat_title_only",
    MaT_TITLE_DESC = "mat_title_desc",
}

export enum RagPipeline {
    HYBRID = "hybrid",
    GRAPH = "graph",
    FUSION = "fusion"
}

export enum Region {
    NORCAL = "Northern California",
    SOCAL = "Southern California",
    OUT_OF_STATE = "Outside California",
    REMOTE = "Remote"
}

export enum TimeFrame {
    ASAP = "As soon as possible",
    SUMMER = "Starting during summer",
    WINTER = "Starting during winter"
}